### PR TITLE
Fix error prone warnings in :core-execution

### PR DESCRIPTION
**Pull request by lptr**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Review by asodja — APPROVED** [review]
LGTM, added a small comment about annotations dependencies

File: platforms/core-execution/execution/build.gradle.kts [hunk 1]
```diff
@@ -4,27 +4,14 @@ plugins {
...
-}
-
 dependencies {
     api(libs.guava)
     api(libs.jsr305)
     api(libs.slf4jApi)
 
     api(projects.concurrent)
     api(projects.javaLanguageExtensions)
+    api(libs.errorProneAnnotations)
```
Comment (asodja) [inline platforms/core-execution/execution/build.gradle.kts:14]: ❓ Can this annotations be `compileOnly`?

Comment (lptr) [reply platforms/core-execution/execution/build.gradle.kts]: Good point, fixed

**Comment by lptr** [comment]
@bot-gradle test this please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/history/changes/ExecutionStateChanges.java [hunk 2]
```diff
@@ -72,11 +64,6 @@ public ImmutableList<String> getChangeDescriptions() {
             public InputChangesInternal createInputChanges() {
                 return new NonIncrementalInputChanges(beforeExecutionState.getInputFileProperties(), incrementalInputProperties);
             }
-
-            @Override
-            public BeforeExecutionState getBeforeExecutionState() {
```
Comment (reinsch82) [inline platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/history/changes/ExecutionStateChanges.java:77]: 👍 I love how code evaporates

**Comment by cobexer** [comment]
@bot-gradle merge

**Pull request merged**